### ISSUE TITLE
PostgreSQL UDT parameter cannot be persisted by DAO class

### ISSUE THREAD

**Issue by vkuzel** [jOOQ/jOOQ#5401]
I have a table with UDT column.

``` sql
CREATE TYPE some_udt AS (
  value INT
);

CREATE TABLE some_table (
  id              BIGSERIAL PRIMARY KEY,
  some_udt_column some_udt NOT NULL
);
```

Then I've generated POJOs and DAOs.

``` xml
<generate>
    <pojos>true</pojos>
    <daos>true</daos>
    ...
</generate>
```

If I try to persist POJO by DAO an exception will occur.

``` java
SomeUdt someUdt = new SomeUdt(12345);

SomeTable someTable = new SomeTable();
someTable.setSomeUdtColumn(someUdt);

someTableDao.insert(someTable);
```

```
Caused by: org.jooq.exception.DataTypeException: Cannot convert from SomeUdt (12345) (class com.jooq.test.udt.pojos.SomeUdt) to class com.jooq.test.udt.records.SomeUdtRecord
    at org.jooq.tools.Convert$ConvertAll.fail(Convert.java:1113) ~[jooq-3.8.3.jar:na]
```

Seems like JOOQ tries to convert `pojos.SomeUdt` class to `records.SomeUdtRecord` class which is not supported by `Convert` class.

Here is the sample application: https://github.com/vkuzel/Jooq-Postgres-Udt-Problem
JOOQ 3.8.3

**Comment by lukaseder** [comment]
Thank you very much for reporting. Will look into this soon!

**Comment by lukaseder** [comment]
Postponing this until we know what we'll do with these DAOs (#5984)

**Comment by chuanhhoang** [comment]
Hi,
When will this be resolved? If it hasn't, is there any workaround for this?

**Comment by lukaseder** [comment]
> When will this be resolved?

Yes

> If it hasn't, is there any workaround for this?

I don't think there is, sorry

**Comment by lukaseder** [comment]
This seems to be a duplicate of https://github.com/jOOQ/jOOQ/issues/6917, which should have been fixed in jOOQ 3.15.0 via these improvements:

- https://github.com/jOOQ/jOOQ/issues/10229
- https://github.com/jOOQ/jOOQ/issues/11183

Closing as duplicate
